src/Aspire.Hosting.Orleans/OrleansService.cs

```diff
@@ -1,34 +1,59 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Aspire.Hosting.ApplicationModel;
+
 namespace Aspire.Hosting.Orleans;
 
 /// <summary>
 /// Describes an Orleans service.
 /// </summary>
-/// <param name="builder">The distributed application builder.</param>
-/// <param name="name">The service name.</param>
-public class OrleansService(IDistributedApplicationBuilder builder, string name)
+public class OrleansService
 {
+    /// <summary>Initializes a new <see cref="OrleansService"/> instance.</summary>
+    /// <param name="builder">The distributed application builder.</param>
+    /// <param name="name">The service name.</param>
+    public OrleansService(IDistributedApplicationBuilder builder, string name)
+    {
+        Name = name;
+        Builder = builder;
+        ServiceId = ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, $"{name}-service-id", secret: false, new GenerateParameterDefault
+        {
+            Lower = true,
+            Upper = false,
+            Special = false,
+            Numeric = true,
+            MinLength = 25
+        });
+        ClusterId = ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, $"{name}-cluster-id", secret: false, new GenerateParameterDefault
+        {
+            Lower = true,
+            Upper = false,
+            Special = false,
+            Numeric = true,
+            MinLength = 25
```

Comment:
```suggestion
            Upper = false,
            Special = false,
            MinLength = 25
        });
        ClusterId = ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, $"{name}-cluster-id", secret: false, new GenerateParameterDefault
        {
            Upper = false,
            Special = false,
            MinLength = 25
```

You don't need to specify the `true`s.

You can also create a helper method for this to reduce the code duplication:

```
ServiceId = CreateGeneratedParameter(builder,  $"{name}-service-id");
ClusterId = CreateGeneratedParameter(builder,  $"{name}-cluster-id");

static ParameterResource CreateGeneratedParameter(IDistributedApplicationBuilder builder, string name) => 
ParameterResourceBuilderExtensions.CreateGeneratedParameter(builder, name, secret: false, new GenerateParameterDefault
    {
        Upper = false,
        Special = false,
        MinLength = 25
    });